Controllers/ProjectController.cs

```diff
@@ -14,170 +11,69 @@ namespace ToggleBuddy.API.Controllers
+            var response = await _projectService.GetProjectByIdForCurrentUserAsync(id, User);
+            return HandleApiResponse(response);
         }
 
-
-        // PUT: api/Project/5
-        [HttpPut]
-        [Route("{id:Guid}")]
-        [ValidateModel]
-        public async Task<IActionResult> UpdateProjectById([FromRoute] Guid id, [FromBody] ProjectRequestDto projectRequestDto)
+        // PUT: api/Project/{id}
+        [HttpPut("{id:Guid}")]
+        public async Task<IActionResult> UpdateProjectById(Guid id, [FromBody] ProjectRequestDto projectRequestDto)
         {
-            var currentUser = await _userRepository.GetCurrentUserAsync(User);
-            if (currentUser == null)
-            {
-                _apiResponse.Message = "User not found";
-                _apiResponse.Result = null;
-                _apiResponse.Status = ResponseStatus.Error;
-                return NotFound(_apiResponse);
-            }
-
-            var project = await _projectRepository.GetProjectByIdForCurrentUserAsync(id, currentUser.Id);
-            if (project == null)
-            {
-                _apiResponse.Message = "Project not found";
-                _apiResponse.Result = null;
-                _apiResponse.Status = ResponseStatus.Error;
-                return NotFound(_apiResponse);
-            }
-
-            var updatedProject = _mapper.Map<Project>(projectRequestDto);
-            updatedProject.Id = project.Id;
-
-            var result = await _projectRepository.UpdateProjectAsync(id, updatedProject);
-
-            var projectResponseDto = _mapper.Map<ProjectResponseDto>(result);
-
-            _apiResponse.Result = projectResponseDto;
-            _apiResponse.Message = "Project updated successfully";
-
-            return Ok(_apiResponse);
+            var response = await _projectService.UpdateProjectAsync(id, projectRequestDto, User);
+            return HandleApiResponse(response);
         }
 
-        // DELETE: api/Project/5
-        [HttpDelete]
-        [Route("{id:Guid}")]
-        public async Task<IActionResult> DeleteProjectById([FromRoute] Guid id)
+        // DELETE: api/Project/{id}
+        [HttpDelete("{id:Guid}")]
+        public async Task<IActionResult> DeleteProjectById(Guid id)
         {
-            var currentUser = await _userRepository.GetCurrentUserAsync(User);
-            if (currentUser == null)
-            {
-                _apiResponse.Message = "User not found";
-                _apiResponse.Result = null;
-                _apiResponse.Status = ResponseStatus.Error;
-                return NotFound(_apiResponse);
-            }
+            var response = await _projectService.DeleteProjectAsync(id, User);
+            return HandleApiResponse(response);
+        }
 
-            var project = await _projectRepository.GetProjectByIdForCurrentUserAsync(id, currentUser.Id);
-            if (project == null)
+        private IActionResult HandleApiResponse<T>(ApiResponse<T> response)
```

Comment:
Might this method be useful to other controllers. Maybe it should be a central utility method any controller can use

Reply:
I'd extract it out to make it reusable.